packages/frontpage/lib/data/db/vote.ts

```diff
@@ -40,3 +41,214 @@ export const getVoteForComment = cache(async (commentId: number) => {
 
   return rows[0] ?? null;
 });
+
+export type UnauthedCreatePostVoteInput = {
+  repo: DID;
+  rkey: string;
+  hydratedVoteRecordValue: {
+    createdAt: string;
+    subject: {
+      cid: string;
+      uri: {
+        rkey: string;
+        value: string;
+        authority: string;
+        collection: string;
+      };
+    };
+  };
+  hydratedRecord: {
+    cid: string;
+  };
+};
+
+export const unauthed_createPostVote = async ({
+  repo,
+  rkey,
+  hydratedVoteRecordValue,
+  hydratedRecord,
+}: UnauthedCreatePostVoteInput) => {
+  await db.transaction(async (tx) => {
+    const subject = (
+      await tx
+        .select()
+        .from(schema.Post)
+        .where(eq(schema.Post.rkey, hydratedVoteRecordValue.subject.uri.rkey))
+    )[0];
+
+    if (!subject) {
+      throw new Error(
+        `Subject not found with uri: ${hydratedVoteRecordValue.subject.uri.value}`,
+      );
+    }
+
+    if (subject.authorDid === repo) {
+      throw new Error(`[naughty] Cannot vote on own content ${repo}`);
+    }
+    await tx.insert(schema.PostVote).values({
+      postId: subject.id,
+      authorDid: repo,
+      createdAt: new Date(hydratedVoteRecordValue.createdAt),
+      cid: hydratedRecord.cid,
+      rkey,
+    });
+
+    await tx
+      .update(schema.PostAggregates)
+      .set({
+        voteCount: sql`${schema.PostAggregates.voteCount} + 1`,
+      })
+      .where(eq(schema.PostAggregates.postId, subject.id));
+
+    await tx.update(schema.PostAggregates).set({
+      rank: sql<number>`
+                (CAST(COALESCE(${schema.PostAggregates.voteCount}, 1) AS REAL) / (pow((JULIANDAY('now') - JULIANDAY(${schema.PostAggregates.createdAt})) * 24 + 2,1.8)))`,
+    });
+  });
+};
+
+export type UnauthedCreateCommentVoteInput = {
+  repo: DID;
+  rkey: string;
+  hydratedVoteRecordValue: {
```

Comment:
Calling this `vote` would be consistent to the equivalent comment and post methods. It should also reference the `atproto.Vote` type instead of being copied here.